deployments/helm/nvidia-device-plugin/templates/daemonset-gfd.yml

```
@@ -57,7 +57,31 @@ spec:
       {{- end }}
       {{- if $options.hasConfigMap }}
       shareProcessNamespace: true
+      {{- end }}
       initContainers:
+      - image: {{ include "nvidia-device-plugin.fullimage" . }}
+        name: gpu-feature-discovery-imex-init
+        command:
+        - "/bin/bash"
+        - "-c"
+        - |
+          if [[ ! -d /driver-root/etc/nvidia-imex ]]; then
```

Question: How would this look like in the GPU Operator, where driver root could be `/driver-root` OR `/host`? Would we simply check both locations for the presence of `/etc/nvidia-imex`?

---

Shouldn't we be using this?

```suggestion
          if [[ ! -d {{ .Values.nvidiaDriverRoot }}/etc/nvidia-imex ]]; then
```

---

No. The container path will always be `/driver-root`.

---

@tariq1890 no, we mount either `{{ .Values.nvidiaDriverRoot }}` or `/` to `/driver-root` in the container. This means that in the container we should always use `/driver-root`.

---

My point is that in the operator use case, we will mount both `/` --> `/host` and `/run/nvidia/driver` --> `/driver-root`. I am assuming we need to update this script in the operator to look for the `/etc/nvidia-imex` directory at both locations? Or just reference the `DRIVER_ROOT_CTR_PATH` variable...

---

Note that the code still checks both `${GFD_IMEX_NODES_CONFIG_FILE}` (with a default value of `/etc/nvidia-imex/nodes_config.cfg`) and `${DRIVER_ROOT_CTR_PATH}/${GFD_IMEX_NODES_CONFIG_FILE}`, so as long as the operator mounts the file and sets the `DRIVER_ROOT_CTR_PATH` envvar accordingly it should work.

---

@cdesiniotis since we're using `DRIVER_ROOT_CTR_PATH` things should work since the validator sets this to `/host` for a host-installed driver here: https://github.com/NVIDIA/gpu-operator/blob/040d1539f6a8e2e114a9f4a4141e6e35cc6ca298/validator/main.go#L841

---

Got it.